td-agent/Rakefile

```diff
@@ -941,6 +1083,208 @@ EOS
   end
 end
 
+class CompressDMG
+  include Rake::DSL
+
+  def initialize(version, window_bounds, pkg_position)
+    @version = version
+    @window_bounds = window_bounds
+    @pkg_position = pkg_position
+    @dmg_temp_name = "rw.td-agent-#{@version}.dmg"
+    @dmg_name = "td-agent-#{@version}.dmg"
+    @pkg_name = "td-agent-#{@version}.pkg"
+    @volume_name = "Td-Agent"
+    @device = nil
+    @osascript_path = File.join("resources", "dmg", "td-agent.osascript")
+  end
+
+  def clean_dmgs
+    rm_f(@dmg_name)
+    rm_f(@dmg_temp_name)
+  end
+
+  def clean_disks
+    disks = `mount | grep \"/Volumes/#{@volume_name}\" | awk '{print $1}'`
+    disks.split("\n").each do |disk|
+      disk.chomp!
+
+      sh("hdiutil", "detach", disk)
+    end
+  end
+
+  def create_rw_dmg
+    sh("hdiutil",
+       "create", "-ov",
+       "-srcfolder", "dmg",
+       "-format", "UDRW",
+       "-volname", @volume_name,
+       @dmg_temp_name)
+  end
+
+  def get_attached_rw_dmg
+    @device = `hdiutil attach -readwrite -noverify -noautoopen #{@dmg_temp_name} | egrep '^/dev/' | sed 1q | awk '{print $1}'`.strip
+    sleep 5
```

Comment:
Is this sleep  required?

Reply:
macOS Finder works a but slower than command line script execution...

Reply:
Got it. Thanks!